Dalamud/Game/ClientState/Actors/Types/Actor.cs

```diff
@@ -71,6 +73,86 @@ public class Actor : IEquatable<Actor> {
         /// </summary>
         public virtual int TargetActorID => 0;
 
+        public ObjectVisibility ObjectVisibility => this.actorStruct.ObjectVisibility;
+
         bool IEquatable<Actor>.Equals(Actor other) => this.ActorId == other.ActorId;
+
+        /// <summary>
+        /// Hides the actor
+        /// </summary>
+        public async void Hide()
+        {
+            var player = ObjectKind == ObjectKind.Player;
+
+            unsafe
+            {
+                ObjectKind* kind = (ObjectKind*)(Address + ActorOffsets.ObjectKind);
+                ObjectVisibility* vis = (ObjectVisibility*)(Address + ActorOffsets.ObjectVisibility);
+                {
+                    if (player)
+                    {
+                        *kind = ObjectKind.BattleNpc;
```

Comment:
what you should do instead above is like 

    var kind = *(ObjectKind*)(Address + ActorOffsets.ObjectKind);

then you can just

    kind = ObjectKind.BattleNpc;

Reply:
or just like `var actor = *(Actor*)(Address)` and then `actor.ObjectKind = thing` and etc.

Reply:
doing the first suggestion doesn't work at all, and the second one leads to this compile error (``var actor = *(Structs.Actor*)(Address);``):
```
Cannot take the address of, get the size of, or declare a pointer to a managed type ('Actor')
```

Reply:
ah yeah, the first one wouldn't work, I misread what was there entirely

the second should have actually been something like the following:

    var actor = (Actor*)address;
    actor->ObjectKind = ...

Reply:
Same error, it's on this line, btw:
```
var actor = (Structs.Actor*) Address;
```

Reply:
i give up, its probably something to do with the explicit layout or something stupid, just do it the way you have it lol

Reply:
🥳